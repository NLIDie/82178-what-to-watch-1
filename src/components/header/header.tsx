import * as React from "react";

// Components
import {Logo} from "../logo/logo";

export function Header(): JSX.Element {
  return (
    <header
      className="page-header"
      style={{
        position: `absolute`,
        top: 0,
        left: 0,
        right: 0,
        zIndex: 2
      }}
    >
      <Logo />

      <h1 className="visually-hidden">WTW</h1>

      <div className="user-block">
        <div className="user-block__avatar">
          <img
            src="../src/assets/img/avatar.jpg"
            alt="User avatar"
            width="63"
            height="63"
          />
        </div>
      </div>
    </header>
  );
}
