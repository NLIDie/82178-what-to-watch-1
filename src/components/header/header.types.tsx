export type TProps = {
  userAvatar: string;
  isAuthorized: boolean;
};
