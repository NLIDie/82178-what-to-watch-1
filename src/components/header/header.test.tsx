import * as React from "react";
import {create} from "react-test-renderer";

// Components
import {Header} from "./header";

describe(`Components/Header`, () => {
  it(`Correct rendering the component`, () => {
    const tree = create(<Header />);

    expect(tree).toMatchSnapshot();
  });
});
