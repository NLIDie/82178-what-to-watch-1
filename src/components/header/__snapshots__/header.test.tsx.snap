// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/header Correct rendering the component 1`] = `
<header
  className="page-header"
  style={
    Object {
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 2,
    }
  }
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
    >
      <img
        alt="User avatar"
        height="63"
        src="../src/assets/img/avatar.jpg"
        width="63"
      />
    </div>
  </div>
</header>
`;
